AbilityChanger/Abilities/DoubleJump.cs

```diff
@@ -9,19 +9,58 @@ public class DoubleJump : AbilityManager {
         public DoubleJump() : base (){
             On.HeroController.DoubleJump += DoubleJumpFixedUpdate;
             On.HeroController.DoDoubleJump += DoubleJumpAbilityTrigger;
+            On.HeroController.CancelDoubleJump += CancelComplete;
         }
+
+        private void CancelComplete(On.HeroController.orig_CancelDoubleJump orig, HeroController self)
+        {
+            if (HeroControllerR.doubleJump_steps > 3) HandleComplete();
+            else if(HeroControllerR.doubleJump_steps!=0) HandleCancel();
+            orig(self);
+        }
+
         public void DoubleJumpAbilityTrigger(On.HeroController.orig_DoDoubleJump orig, HeroController self){
+            #region Trigger
+            if (currentAbility.hasTrigger()) { HandleTrigger(); return; }
+            #endregion
+
+            #region Start
+            HandleStart();
             orig(self);
-            if(isCustom()){
-                this.handleAbilityUse();
-            }
+            #endregion 
         }
         public void DoubleJumpFixedUpdate(On.HeroController.orig_DoubleJump orig, HeroController self){
-            if(!isCustom()){
-                orig(self);
-            }
+            #region OnGoing
+            if (currentAbility.hasOngoing()) {  HandleOngoing();  }
+            #endregion 
+            orig(self);
```

Comment:
Should making hasOngoing()= true mean that we don't do orig(self) ?  and let the new ability handle it if it is a custom ability? 

I think we should to allow more flexibility in what can be done.